common/lib/capa/capa/responsetypes.py

```diff
@@ -1266,19 +1500,70 @@ def get_score(self, student_answers):
                         tolerance=float_info.epsilon,
                         relative_tolerance=True
                 ):
-                    correct = inclusion
+                    is_correct = inclusion
                     break
             else:
-                correct = boundaries[0] < student_float < boundaries[1]
+                if boundaries[0] < student_float < boundaries[1]:
+                    is_correct = 'correct'
+                else:
+                    if credit_type is False:
+                        pass
+                    elif 'close' in credit_type:
+                        """
+                        Partial credit: 50% if the student is outside the specified boundaries,
+                        but within an extended set of boundaries.
+                        """
+                        extended_boundaries = []
+                        boundary_range = boundaries[1]-boundaries[0]
+                        extended_boundaries.append(boundaries[0] - partial_range * boundary_range)
+                        extended_boundaries.append(boundaries[1] + partial_range * boundary_range)
+                        if extended_boundaries[0] < student_float < extended_boundaries[1]:
+                            is_correct = 'partially-correct'
+                    
         else:
             correct_float = self.get_staff_ans(self.correct_answer)
-            correct = compare_with_tolerance(
-                student_float, correct_float, self.tolerance
-            )
-        if correct:
-            return CorrectMap(self.answer_id, 'correct')
+            
+            """
+            Partial credit is available in three cases:
+            - If the student answer is within expanded tolerance of the actual answer,
+              the student gets 50% credit. (Currently set as the default.)
+              Set via partial_credit="close" in the numericalresponse tag.
+              
+            - If the student answer is within regular tolerance of an alternative answer, 
+              the student gets 50% credit. (Same default.)
+              Set via partial_credit="list"
+              
+            - If the student answer is within expanded tolerance of an alternative answer,
+              the student gets 25%. (We take the 50% and square it, at the moment.)
+              Set via partial_credit="list,close" or "close, list" or the like.
+            """
```

Comment:
Don't tear your hair out. If it's not possible, que sera.

On Tue, Jun 23, 2015 at 4:06 PM, Colin-Fredericks <notifications@github.com>
wrote:

> In common/lib/capa/capa/responsetypes.py
> <https://github.com/edx/edx-platform/pull/8574#discussion_r33085286>:
>
> > -            return CorrectMap(self.answer_id, 'correct')
> > +
> > +            """
> > +            Partial credit is available in three cases:
> > +            - If the student answer is within expanded tolerance of the actual answer,
> > +              the student gets 50% credit. (Currently set as the default.)
> > +              Set via partial_credit="close" in the numericalresponse tag.
> > +
> > +            - If the student answer is within regular tolerance of an alternative answer,
> > +              the student gets 50% credit. (Same default.)
> > +              Set via partial_credit="list"
> > +
> > +            - If the student answer is within expanded tolerance of an alternative answer,
> > +              the student gets 25%. (We take the 50% and square it, at the moment.)
> > +              Set via partial_credit="list,close" or "close, list" or the like.
> > +            """
>
> Will change comment style. If I can break out the function without tearing
> my hair out I will.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8574/files#r33085286>.
>